### PR TITLE
LF: use constance for language version.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
We use the constants defined in LanguageVersion/TransactionVersions instead repeating the
somewhat magic constants "6", "7", "8", "10" and  "dev" in multiple places?

This is part of polishing #7788

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala [hunk 1]
```diff
@@ -1935,37 +1935,37 @@ class EngineTest
...
         Traversable.empty,
         Traversable.empty,
-        LV(LV.Major.V1, v),
+        version,
         None
       )
 
     "reject disallow packages" in {
       val negativeTestCases = Table(
         ("pkg version", "minVersion", "maxVertion"),
```
Comment (hurryabit) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:1957]:
```suggestion
        ("pkg version", "minVersion", "maxVersion"),
```

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala [hunk 2]
```diff
@@ -1935,37 +1935,37 @@ class EngineTest
...
-        (LV.Minor("7"), LV.Minor("6"), LV.Minor("8")),
-        (LV.Minor("8"), LV.Minor("6"), LV.Minor("8")),
-        (LV.Minor("dev"), LV.Minor("6"), LV.Minor("dev")),
+        (LV.v1_6, LV.v1_6, LV.v1_8),
+        (LV.v1_7, LV.v1_6, LV.v1_8),
+        (LV.v1_8, LV.v1_6, LV.v1_8),
+        (LV.v1_dev, LV.v1_6, LV.v1_dev),
       )
       val positiveTestCases = Table(
         ("pkg version", "minVersion", "maxVertion"),
```
Comment (hurryabit) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:1964]:
```suggestion
        ("pkg version", "minVersion", "maxVersion"),
```

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionTimeline.scala [hunk 3]
```diff
@@ -168,12 +168,9 @@ object VersionTimeline {
   }
 
   private[lf] val stableLanguageVersions =
-    VersionRange(
-      min = LanguageVersion(Major.V1, Minor("6")),
-      max = LanguageVersion(Major.V1, Minor("8")),
-    )
+    VersionRange(min = LanguageVersion.v1_6, max = LanguageVersion.v1_8)
```
Comment (hurryabit) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionTimeline.scala:171]: Don't we have a constant for "maximum stable version"? Shouldn't we use that here as the upper bound instead?

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/VersionTimelineSpec.scala [hunk 4]
```diff
@@ -73,7 +73,8 @@ class VersionTimelineSpec
       inside(inAscendingOrder.last) {
         case Both(
             Both(ValueVersion("dev"), TransactionVersion("dev")),
```
Comment (hurryabit) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/VersionTimelineSpec.scala:75]: We should probably do the same for value and transactions versions as well at some point.

**Review by hurryabit — APPROVED** [review]
Very nice. Thank you.

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionTimeline.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionTimeline.scala:171]: I will drop soon the `VersionTimeline` so we should not worry too much.
I will try to take this consideration when cleaning further `LanguageVersions`.

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/VersionTimelineSpec.scala (hunk 4 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/VersionTimelineSpec.scala:75]: :+1:

Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/VersionTimelineSpec.scala]: I include it in the PR.

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionTimeline.scala (hunk 3 above)
Comment (hurryabit) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionTimeline.scala:171]: Cool. Thanks.

**Pull request merged**